087-external-certificate-manager.md

```diff
@@ -0,0 +1,196 @@
+
+### Handling trust rollout
+
+Similar to today Strimzi will use the notion of a "generation" to determine whether to roll the cluster to pick up changes in either the `<CLUSTER_NAME>-cluster-ca-cert` or `<CLUSTER_NAME>-clients-ca-cert`.
+When the user creates the CA cert Secret they must add the `strimzi.io/ca-cert-generation` annotation to it.
+If the user updates the Secret to change the certificates included they must increment the annotation to inform Strimzi it has changed.
+Similar to today Strimzi will put the annotation on the pods (Kafka, ZooKeeper etc) to be able to spot when the generation has been changed.
+
+### Issuing certificates
+
+If the user has enabled this feature, when Strimzi needs to issue a certificate, instead of using the existing internal mechanism it will create a `Certificate` custom resource.
+Strimzi will wait during the reconciliation loop for the `Certificate` status to indicate that the certificate has been issued before continuing.
+When issuing cluster certificates (e.g for Kafka etc), once the certificate has been issued, Strimzi will annotate the cert-manager provided Secret with the `strimzi.io/server-cert-hash` annotation with the value being the hash of the certificate in the Secret.
+Similar to today, Strimzi will also add this hash annotation to the pods to track whether they are mounting the latest version of the Secret.
+
+### Tracking changes to cluster end-entity certificates
+
+Cert-manager will be responsible for renewing all end-entity certificates.
+When a certificate is renewed cert-manager will update the related Secret.
+
+For cluster certificates (e.g. for Kafka etc), Strimzi will track and handle these changes using the `strimzi.io/server-cert-hash` annotation.
+During the reconciliation loop, even if all cluster end-entity certificates have been issued, Strimzi will patch the certificate Secrets with the correct `strimzi.io/server-cert-hash` annotation.
+The value of this annotation can then be compared with the value on the pods to determine whether the pods need to be restarted to pick up a new Secret.
+
+For user certificates (issued by the User Operator), the user will be responsible for making sure their applications notice cert-manager renewing the certificates and are updated to use the new certificate.
+
+## Affected/not affected projects
+
+This affects the Cluster Operator and User Operator.
+
+## Compatibility
+
+This feature will be optional and not disabled by default.
+
+### Migrating to this feature
+
+To start using this feature in an existing Kafka cluster the user must:
+1. Install cert-manager and create an `Issuer`.
+2. Pause reconciliation for their Kafka cluster.
+3. Update the `<CLUSTER_NAME>-cluster-ca-cert` and/or `<CLUSTER_NAME>-clients-ca-cert` Secrets to:
+   1. contain the CA(s) for the `Issuer` (keeping the old CA cert in the Secret still)
```

Comment:
I think this part is really important - `(keeping the old CA cert in the Secret still)` - and should be made more prominent? We should also cover the point when it should be removed.